packages/aws-cdk-lib/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -746,6 +775,11 @@ export class FlowLog extends FlowLogBase {
       }).join(' ');
     }
 
+    let trafficType: FlowLogTrafficType | undefined = undefined;
+    if (!['TransitGateway', 'TransitGatewayAttachment'].includes(props.resourceType.resourceType)) {
```

Comment:
As to the code, because I had intended behavior that already exists will not changed.

OK, I will change as you say. But I don't know how should I write well so [I asked you](https://github.com/aws/aws-cdk/pull/27262#discussion_r1401639014).